FirebaseTestingSupport/Database/Sources/FIRDataSnapshotFake.m

```diff
@@ -0,0 +1,28 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "FirebaseTestingSupport/Database/Sources/Public/FirebaseDatabaseTestingSupport/FIRDataSnapshotFake.h"
+
+@implementation FIRDataSnapshotFake
+
+- (instancetype)init {
```

Comment:
If the warning is expected "Designated initializer missing a 'super' call to a designated initializer of the super class", please disable with a pragma. See other examples in the repo `git grep pragma`.